test/addIndex.js

```diff
@@ -0,0 +1,74 @@
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('addIndex', function() {
+  describe('works with unary functions like `map`', function() {
+    var times2 = function(x) {return x * 2;};
+    var addIndexParam = function(x, idx) {return x + idx;};
+    var squareEnds = function(x, idx, list) {
+      return (idx === 0 || idx === list.length - 1) ? x * x : x;
+    };
+    var mapIndexed = R.addIndex(R.map);
+
+    it('working just like a normal map', function() {
+      assert.deepEqual(mapIndexed(times2, [1, 2, 3, 4]), [2, 4, 6, 8]);
+    });
+
+    it('passing the index as a second parameter to the callback', function() {
+      assert.deepEqual(mapIndexed(addIndexParam, [8, 6, 7, 5, 3, 0, 9]), [8 + 0, 6 + 1, 7 + 2, 5 + 3, 3 + 4, 0 + 5, 9 + 6]);
```

Comment:
I remember having a landline with a seven-digit phone number, and that was only after a digit was added to our six-digit number. :)